pageserver/src/tenant/timeline.rs

```diff
@@ -2218,6 +2258,36 @@ impl Timeline {
                     Ok(timeline) => timeline,
                     Err(e) => return Err(PageReconstructError::from(e)),
                 };
+
+                // It's possible that the ancestor timeline isn't active yet, or
+                // is active but hasn't yet caught up to the branch point. Wait
+                // for it.
+                //
+                // This cannot happen while the pageserver is running normally,
+                // because you cannot create a branch from a point that isn't
+                // present in the pageserver yet. However, we don't wait for the
+                // branch point to be uploaded to cloud storage before creating
+                // a branch. Therefore, after a (non-graceful) pageserver
+                // restart or tenant attach, it is possible that the branch
+                // point wasn't flushed to disk or uploaded to remote storage
+                // yet nor re-streamed from the safekeepers. In other words,
+                // it's possible that the branch point is beyond the ancestor's
+                // last-record-LSN.
+                match ancestor.wait_to_become_active(ctx).await {
+                    Ok(()) => {}
+                    Err(state) if state == TimelineState::Stopping => {
+                        return Err(PageReconstructError::Stopping);
+                    }
+                    Err(state) => {
+                        return Err(PageReconstructError::Other(anyhow::anyhow!(
+                            "Timeline {} will not become active. Current state: {:?}",
+                            self.timeline_id,
+                            &state,
+                        )));
+                    }
+                }
```

Comment:
1. I think the `Stopping` variant should be called `AncestorStopping`.
2. It should contain the ancestor timeline id that is stopping, and reproduce that timeline id in the error message.
3. The `Other(anyhow...())` message is using `self.timeline_id`, but, actually, it should use `ancestor.timeline_id`.